sdlib/d/rt/trampoline.d

```diff
@@ -53,7 +59,20 @@ void* runThread(ThreadRunner* runner) {
 	auto fun = runner.fun;
 	auto arg = runner.arg;
 
-	createThread();
+	createThread(false);
+	__sd_gc_free(runner);
+
+	// Make sure we clean up after ourselves.
+	scope(exit) destroyThread();
+
+	return fun(arg);
+}
+
+void* runGCThread(ThreadRunner* runner) {
+	auto fun = runner.fun;
+	auto arg = runner.arg;
+
+	createThread(true);
```

Comment:
Lots of duplication here. This calls for a common implementation.

Reply:
Moved to template with bool parameter.